### PR TITLE
Add PHPStan updates

### PR DESCRIPTION
**Pull request by muglug**

### CODE REVIEW

File: FunctionSignatureMap.php [hunk 1]
```diff
@@ -9943,6 +9943,7 @@
 'simplexml_load_file' => ['SimpleXMLElement|false', 'filename'=>'string', 'class_name='=>'string', 'options='=>'int', 'ns='=>'string', 'is_prefix='=>'bool'],
 'simplexml_load_string' => ['SimpleXMLElement|false', 'data'=>'string', 'class_name='=>'string', 'options='=>'int', 'ns='=>'string', 'is_prefix='=>'bool'],
 'SimpleXMLElement::__construct' => ['void', 'data'=>'string', 'options='=>'int', 'data_is_url='=>'bool', 'ns='=>'string', 'is_prefix='=>'bool'],
+'SimpleXMLElement::__get' => ['SimpleXMLElement', 'name'=>'string'],
```
Comment (TysonAndre) [inline FunctionSignatureMap.php]: This implements magic getters internally, but not through __get(). DynamicPropertyMap is what Phan uses to keep track of this type of edge case.

```
php > $x = new SimpleXMLElement('<b>x</b>');
php > $x->__get('b');

Warning: Uncaught Error: Call to undefined method SimpleXMLElement::__get() in php shell code:1
Stack trace:
#0 {main}
  thrown in php shell code on line 1
php > var_export($x->b);
SimpleXMLElement::__set_state(array(
))
```

The rest of the changes look good

Comment (muglug) [reply FunctionSignatureMap.php]: Just removed that line

File: FunctionSignatureMap.php [hunk 2]
```diff
@@ -9859,7 +9859,7 @@
 'SessionUpdateTimestampHandler::validateId' => ['char', 'id'=>'string'],
 'SessionUpdateTimestampHandlerInterface::updateTimestamp' => ['bool', 'key'=>'string', 'val'=>'string'],
 'SessionUpdateTimestampHandlerInterface::validateId' => ['bool', 'key'=>'string'],
-'set_error_handler' => ['?string', 'error_handler'=>'null|callable(int,string,string,int,array):bool|callable(int,string,string,int):bool|callable(int,string,string):bool|callable(int,string):bool', 'error_types='=>'int'],
+'set_error_handler' => ['?callable', 'error_handler'=>'null|callable(int,string,string=,int=,array=):bool', 'error_types='=>'int'],
```
Comment (TysonAndre) [inline FunctionSignatureMap.php:9862]: I overlooked the changes to the parameter union type - Those will need to be changed due to differences in Phan/Psalm's casting rules for callables, as discussed earlier.

The change to the return type makes sense.

Comment (TysonAndre) [reply FunctionSignatureMap.php:9862]: The fix was committed to master.

Comment (muglug) [reply FunctionSignatureMap.php:9862]: Awesome, sorry

**Pull request merged**